appservice/src/TunnelProxy.ts

```diff
@@ -59,64 +57,59 @@ class TunnelSocket extends EventEmitter {
             this.emit('error', err);
         });
 
-        this._wsClient.on('connect', (connection: websocket.connection) => {
+        this._websocket = new ws.WebSocket(
```

Comment:
Rather than all this, would it be possible to use `ws.createWebSocketStream` and tie up the streams to the input `Socket` object?

Reply:
Yes: #1109